File: q1582/q1582.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        r, c = len(mat), len(mat[0])
        rowRecord = [0] * r
        colRecord = [0] * c
        for i in range(r): 
            for j in range(c): 
                if mat[i][j] == 1:
                    rowRecord[i] += 1
                    colRecord[j] += 1
        
        res = 0
        for i in range(r): 
            for j in range(c): 
                if mat[i][j] == 1 and rowRecord[i] == 1 and colRecord[j] == 1:
                    res += 1
        return res

File: q1582/q1582.cpp
class Solution
{
public:
    int numSpecial(vector<vector<int>> &mat)
    {
        int r = mat.size();
        int c = mat[0].size();
        vector<int> rowRecord(r, 0);
        vector<int> colRecord(c, 0);
        for (int i = 0; i < r; i++)
        {
            for (int j = 0; j < c; j++)
            {
                if (mat[i][j] == 1)
                {
                    rowRecord[i]++;
                    colRecord[j]++;
                }
            }
        }

        int res = 0;
        for (int i = 0; i < r; i++)
        {
            for (int j = 0; j < c; j++)
            {
                if (mat[i][j] == 1 && rowRecord[i] == 1 && colRecord[j] == 1)
                {
                    res++;
                }
            }
        }
        return res;
    }
};